cmd/krew/cmd/search_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"testing"
+)
+
+func Test_searchByNameAndDesc(t *testing.T) {
+	keyword := "config"
+	tests := []struct {
+		name  string
+		desc  string
+		isHit bool
+	}{
+		{
+			name:  "ctx",
+			desc:  "Switch between contexts in your kubeconfig",
+			isHit: true,
+		},
+		{
+			name:  "kc",
+			desc:  "Interactive CRUD operations to manage kubeconfig",
+			isHit: true,
+		},
+		{
+			name:  "ns",
+			desc:  "Switch between Kubernetes namespaces",
+			isHit: false,
+		},
+	}
+
+	nameData := make([]string, len(tests))
+	descData := make([]string, len(tests))
+	expectHitCnt := 0
+	for _, tt := range tests {
+		nameData = append(nameData, tt.name)
+		descData = append(descData, tt.desc)
+		if tt.isHit {
+			expectHitCnt++
+		}
+	}
+
+	searchResults := searchByNameAndDesc(keyword, nameData, descData)
```

Comment:
Perhaps a better way to structure this test func would be to:

- declare testPlugins slice
- declare scenarios and expected plugin match list
- run scenarios (using t.Run subtests) and compare expected vs got


This way you can test things explicitly like:

- positive cases
- negative cases (doesn't match)
- fuzzy cases (name matches outrank description matches)